packages/tailwindcss-language-server/src/version-guesser.ts

```diff
@@ -0,0 +1,48 @@
+export type TailwindVersion = '3' | '4'
+
+/**
+ * Determine the likely Tailwind version used by the given file
+ *
+ * This returns an array of possible versions, as a file could contain
+ * features that make determining the version ambiguous.
+ *
+ * The order *does* matter, as the first item is the most likely version.
+ */
+export function guessTailwindVersion(content: string): TailwindVersion[] {
+  if (!content) return []
+
+  // It's likely this is a v4 file if it has a v4 import:
+  // - `@import "tailwindcss"`
+  // - `@import "tailwindcss/theme"
+  // - etc…
+  let HAS_V4_IMPORT = /@import\s*['"]tailwindcss(?:\/[^'"]+)?['"]/
+  if (HAS_V4_IMPORT.test(content)) return ['4']
+
+  // It's likely this is a v4 file if it has a v4-specific feature:
+  // - @theme
+  // - @plugin
+  // - @utility
+  // - @variant
+  // - @custom-variant
+  let HAS_V4_DIRECTIVE = /@(theme|plugin|utility|custom-variant|variant)\s*[^;{]+[;{]/
```

Comment:
I think we can add `@reference` to this list as well

Reply:
OH… yep yep yep thanks good call